core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -183,6 +183,17 @@ public void initialize(String name, Map<String, String> unresolved) {
     String credential = props.get(OAuth2Properties.CREDENTIAL);
     String scope = props.getOrDefault(OAuth2Properties.SCOPE, OAuth2Properties.CATALOG_SCOPE);
     Map<String, String> optionalOAuthParams = OAuth2Util.buildOptionalParam(props);
+    if (!props.containsKey(OAuth2Properties.OAUTH2_SERVER_URI)) {
```

Comment:
This would be a problem for sigv4 users. The Sigv4 code path is currently hidden in the HTTPClient with:

```
private static final String SIGV4_ENABLED = "rest.sigv4-enabled";
  private static final String SIGV4_REQUEST_INTERCEPTOR_IMPL =
      "org.apache.iceberg.aws.RESTSigV4Signer";
```

So we should at least check this property and not log warning if the flag is set to true.

I can do some refactoring to bring them to public after this PR is merged, by formally introduce an additional property like `auth-mode`, that can be of values like `oauth2`, `amazon-sigv4`, and defaults to `oauth2`. And we only check this property if the auth mode is oauth2 or null.

Reply:
Good point

Reply:
But we can do the `auth-mode` change in the later "milestone"

Reply:
@jackye1995 can you take a look at the updated `if` condition?